bundles/paikkatietoikkuna/coordinateconversion/components/table.js

```diff
@@ -0,0 +1,191 @@
+                    me.incrementNumRows();
+                }
+
+                return this.getContainer();
+        },
+        incrementNumRows: function () {
+                this.getContainer().find("#num").text( this.numrows++ );
+        },
+        decrementNumRows: function () {
+                this.getContainer().find("#num").text( this.numrows-- );
+        },
+        isEditable: function ( editable ) {
+            var rows = this.getElements().rows;
+            if( !editable ) {
+                rows.each( function ( row ) {
+                    jQuery(this).find('td').attr("contenteditable", false);
+                });
+            }
+            else {
+                rows.each( function ( row ) {
+                    jQuery(this).find('td').attr("contenteditable", true);
+                });
+            }
+        },
+        removeRow: function () {
+            this.getElements().rows.first().remove();
+            this.decrementNumRows();
+        },
+        /**
+         * @method populate
+         *
+         * {Object} data, each key need to have property lon & lat 
+         */
+        populate: function( cell, data ) {
+            var table = this.getElements().table;
+            var i = 0;
+            for ( var key in data ) {
+                if ( i <= 9 ) {
+                    if ( data.hasOwnProperty( key ) ) {
+                        var row = this.template.row( { coords: data[key] } );
+                        jQuery( cell ).parent().after( row );
+                        this.incrementNumRows();
+                        i++;
+                    }
+                }
+            }
+            table.trigger('rowCountChanged');
+        },
+        addRows: function (coords) {
+            var table = this.getElements().table;
+            for (var i = 0; i < coords.length; i++ ) {
+                var row = this.template.row( { coords: coords[i] } );
+                table.find('tr:first').after(row);
+                this.incrementNumRows();
+            }
+            table.trigger('rowCountChanged');
+        },
+        clearRows: function () {
+            var cells = this.getContainer().find('tr .cell:not(.cell.control)');
+            for(var i = 0; i < cells.length; i++) {
+                var trimmedCellValue = jQuery.trim( jQuery( cells[i] ).html() );
+                if( trimmedCellValue !== "" ) {
+                    jQuery( cells[i] ).parent().remove();
+                    this.decrementNumRows();
+                }
+            }
+        },
+        displayNumOfRows: function () {
+            var me = this;
+            var table = this.getElements().table;
+            table.bind('rowCountChanged', function (evt) {
+                var rows = me.getElements().rows;
+                    if( rows.length >= 1000 ) {
+                        rows.slice(1000).css('display', 'none');
```

Comment:
Please find another way of doing this. Like removing excess rows when new ones are added. This seems like a very poor solution for the problem of having too many rows on the table.